engine/channels.go

```diff
@@ -9,6 +9,57 @@ import (
 	"github.com/onflow/flow-go/model/flow"
 )
 
+// init is called first time this package is imported.
+// It creates and initializes the channel ID map.
+func init() {
+	initializeChannelIdMap()
+}
+
+// channelIdMap keeps a map between channel IDs and list of flow roles involved in that channel ID.
+var channelIdMap map[string]flow.RoleList
+
+// RolesByChannelID returns list of flow roles involved in the channelID.
+func RolesByChannelID(channelID string) (flow.RoleList, bool) {
+	roles, ok := channelIdMap[channelID]
+	return roles, ok
+}
+
+// ValidChannelID returns true if channel ID is a valid channel, i.e.,
+// it is either a cluster channel id or is registered in the channel id map.
+func ValidChannelID(channelID string) bool {
+	if _, ok := RolesByChannelID(channelID); ok {
```

Comment:
I would suggest that the `RolesByChannelID` function should make a check if the channelID is a cluster based channel ID and according remove the cluster suffix and return the correct role. Otherwise, there is an inherent assumption that the `RolesByChannelID` function only works with non-cluster based channel IDs while the name seem to suggest that it will work for any ChannelID.

Reply:
tl;dr - `RolesByChannelID` should call `IsClusterChannelID`

Reply:
Addressed in https://github.com/onflow/flow-go/pull/83/commits/282f02e6a0a33cc6b1815024216cdc70c4128839.